### ISSUE TITLE
recursive import (with usingnamespace) break the zls

### ISSUE THREAD

**Issue by codehz** [zigtools/zls#123]
log: `Connection to server got closed. Server will restart.` (and keep restarting)

recursive import:
file name A.zig and its contents is `usingnamespace @import("./A.zig");`

Expected behavior: print error but don't crash

**Comment by codehz** [comment]
(also crash for 
```zig
fn Func(comptime T: T)
                    ^ when I try to type this
```

**Comment by alexnask** [comment]
Thanks for the report, I will fix this second one in a bit.  
I don't think import cycles should cause infinite loops but I will check it out, there are probably some edgecases where they do.

**Comment by codehz** [comment]
> Thanks for the report, I will fix this second one in a bit.
> I don't think import cycles should cause infinite loops but I will check it out, there are probably some edgecases where they do.

(I modified the issue
only usingnamespace with recusive import will be cause crash

**Comment by alexnask** [comment]
This should be closed in a0f1795f2dfea3913968489072f28473eadce0f3  
I tried it with a couple of examples, I would be grateful if you can try it with the testcase you were using as well.
